[Form] Bootstrap 5 documentation (#15158)
symfony/symfony-docs: form/bootstrap5.rst

Checkboxes and Radios
---------------------

For a checkbox/radio field, calling ``form_label()`` doesn't render anything.
Due to Bootstrap internals, the label is already rendered by ``form_widget()``.

Inline Checkboxes and Radios
----------------------------

If you want to render your checkboxes or radios fields `inline`_, you can add
the `checkbox-inline` or `radio-inline` class, depending of your Symfony Form
type or `ChoiceType` configuration, to the label class.

.. configuration-block::

    .. code-block:: php

        $builder
            ->add('myCheckbox', CheckboxType::class, [
                'label_attr' => [
                    'class' => '`checkbox-inline',
                ],
            ])
            ->add('myRadio', RadioType::class, [
                'label_attr' => [
                    'class' => 'radio-inline',
                ],
            ]);

    .. code-block:: twig

        {{ form_row(form.myCheckbox, {
            label_attr: {
                class: 'checkbox-inline'
            }
        }) }}

        {{ form_row(form.myRadio, {
            label_attr: {
                class: 'radio-inline'
            }
        }) }}

Switches
________

> Bootstrap 5 allow to render checkboxes as "`switches`_". You can enable this feature

Reviewer: ```suggestion
Bootstrap 5 allows to render checkboxes as "`switches`_". You can enable this feature
```

on your Symfony Form ``CheckboxType`` by adding the ``checkbox-switch`` class to the label:

.. configuration-block::

    .. code-block:: php

        $builder->add('myCheckbox', CheckboxType::class, [
            'label_attr' => [
                'class' => 'checkbox-switch',
            ],
        ]);

    .. code-block:: twig

        {{ form_row(form.myCheckbox, {
            label_attr: {
                class: 'checkbox-switch'
            }
        }) }}

.. tip::

    You can also render your switches inline by simply adding the ``checkbox-inline`` class

>     on the ``label_attr`` option.
> 
>     .. code-block:: php

Reviewer 2: ```suggestion
    on the ``label_attr`` option::
```

Author: @OskarStark Thanks for the solution, I had trouble understanding the error

        ...
        'label_attr' => [
            'class' => '`checkbox-inline checkbox-switch',
        ],
        ...

.. caution::

>     Switches only worked with **checkbox**.

Reviewer: ```suggestion
    Switches only work with **checkbox**.
```

Input group
___________

To create `input group`_ in your Symfony Form, simply add the `input-group` class
to the ``row_attr`` option.

.. configuration-block::

    .. code-block:: php

        $builder->add('email', CheckboxType::class, [
            'label' => '@',
            'row_attr' => [
                'class' => 'input-group',
            ],
        ]);

    .. code-block:: twig

        {{ form_row(form.email, {
            label: '@',
            row_attr: {
                class: 'input-group'
            }
        }) }}

.. caution::

>     If you have filled in the `help` option of your form, it will also be rendered

Reviewer: Passive voice is harder to read, especially for non-native speakers, so I would write
```suggestion
    If you fill the `help` option of your form, it will also be rendered
```

    as part of the group.

Floating labels
---------------

To render an input field with a `floating label`_, you must add a ``label``,
a ``placeholder`` and the ``form-floating`` class to the ``row_attr`` option
of your form type.

.. configuration-block::

    .. code-block:: php

        $builder->add('name', TextType::class, [
            'label' => 'Name',
            'attr' => [
                'placeholder' => 'Name',
            ],
            'row_attr' => [
                'class' => 'form-floating',
            ],
        ]);

    .. code-block:: twig

        {{ form_row(form.name, {
            label: 'Name',
            attr: {
                placeholder: 'Name'
            },
            row_attr: {
                class: 'form-floating'
            }
        }) }}

.. caution::

    You **must** provide a ``label`` and a ``placeholder`` to make floating labels

>     works properly.

Reviewer: ```suggestion
    work properly.
```
